src/core/addonManager.js

```diff
@@ -9,8 +9,13 @@ import {
 import { addQueryParams } from 'core/utils';
 
 
-export function hasAddonManager({ navigator } = {}) {
-  return typeof window !== 'undefined' && 'mozAddonManager' in (navigator || window.navigator);
+const testHasWindow = () => typeof window !== 'undefined';
+
+export function hasAddonManager({ hasWindow = testHasWindow, navigator } = {}) {
+  if (!navigator && !hasWindow()) {
+    return undefined;
+  }
+  return 'mozAddonManager' in (navigator || window.navigator);
```

Comment:
It feels like the `if` and this line could be combined as a single return - although that would make for a longer line...